src/common/dataProvider.ts

```diff
@@ -16,60 +14,62 @@ import { DataProvider } from 'ra-core';
  * updateMany          => PUT    http://my.api.url/tables/artists/rows/1,2,3
  * delete              => DELETE http://my.api.url/tables/artists/rows/1
  * deleteMany          => DELETE http://my.api.url/tables/artists/rows/1,2,3
- * 
- * 
- * @example 
- * 
+ *
+ *
+ * @example
+ *
  * import {Admin, Resource} from "react-admin"
  * import { dataProvider } from "./common/dataProvider";
- * 
+ *
  * import {AlbumList} from "./components/Album"
- * 
+ *
  * const primaryKeyDictionary = {
- *    albums: "AlbumId", 
- *    tracks: "TrackId", 
- *    genres: "GenreId", 
+ *    albums: "AlbumId",
+ *    tracks: "TrackId",
+ *    genres: "GenreId",
  *    playlists: "PlayListId"
  * }
- * 
+ *
  * function App() {
  *   return (
- *      <Admin dataProvider={dataProvider(primaryKeyDictionary, "http://my.api.url/tables/")}> 
+ *      <Admin dataProvider={dataProvider(primaryKeyDictionary, "http://my.api.url/tables/")}>
  *        <Resource name="albums" list={AlbumList}/>
  *      </Admin>
  *   )
  * }
  */
 
-export const dataProvider = (pkDictionary: any, apiUrl: string): DataProvider => ({
+export const dataProvider = (
+  pkDictionary: any,
+  apiUrl: string,
+): DataProvider => ({
   getList: (resource: string, params: any) => {
     const { page, perPage } = params.pagination;
     let { field, order } = params.sort;
-  
-    field = field === 'id' ? pkDictionary[resource] : field
+
+    field = field === "id" ? pkDictionary[resource] : field;
     const ordering = order === "ASC" ? `${field}` : `-${field}`;
 
     //since the soul api requires comma separated filters, remove the quotes and curly braces from the filter
     const filter = JSON.stringify(params.filter).replace(/[{} ""]/g, "");
 
-    //Add the _extend query to get many data by using a foreign key 
-    let extend = undefined 
+    //Add the _extend query to get many data by using a foreign key
+    let extend = undefined;
```

Comment:
@AbegaM - can we remove this `chinook` specific code from the `dataProvider`?

Reply:
Sure @IanMayo I will remove it